qa/integration-tests/src/test/java/io/camunda/it/utils/CamundaMultiDBExtension.java

```diff
@@ -282,56 +284,85 @@ private void injectFields(
     }
   }
 
+  private void withRetry(final Callable<Boolean> operation, final int maxAttempt) {
+    int attempt = 0;
+    boolean shouldRetry = true;
+    while (shouldRetry) {
+      try {
+        // if we succeed we don't want to retry
+        shouldRetry = !operation.call();
+      } catch (final Exception ex) {
+        LOGGER.debug(
+            "Failed to execute {}. Attempts: [{}/{}]", operation, attempt + 1, maxAttempt, ex);
+      } finally {
+        // if we reached the max attempt we stop
+        if (++attempt >= maxAttempt) {
+          shouldRetry = false;
+        }
+      }
+
+      if (shouldRetry) {
+        try {
+          // wait a little between retries
+          Thread.sleep(100);
+        } catch (final InterruptedException e) {
+          throw new RuntimeException(e);
+        }
+      }
+    }
+  }
+
   @Override
   public void afterAll(final ExtensionContext context) {
     if (databaseType == DatabaseType.ES || databaseType == DatabaseType.OS) {
       try (final HttpClient httpClient = HttpClient.newHttpClient()) {
+
         // delete indices
         // https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-delete-index.html
-        final URI deleteIndicesEndpoint =
-            URI.create(String.format("%s/%s-*", DEFAULT_ES_URL, testPrefix));
-        HttpRequest httpRequest =
-            HttpRequest.newBuilder().DELETE().uri(deleteIndicesEndpoint).build();
-        HttpResponse<String> response = httpClient.send(httpRequest, BodyHandlers.ofString());
-        int statusCode = response.statusCode();
-        if (statusCode / 100 == 2) {
-          LOGGER.info("Test indices with prefix {} deleted.", testPrefix);
-        } else {
-          LOGGER.warn(
-              "Failure on deleting test indices with prefix {}. Status code: {} [{}]",
-              testPrefix,
-              statusCode,
-              response.body());
-        }
+        withRetry(
```

Comment:
I was first thinking we could use awaitility, but actually I just want to specify some basic attempts and not fail if we reach the max attempt.

Let me know if you have a different idea.